manifests/repo/apt_postgresql_org.pp

```diff
@@ -15,14 +15,15 @@
     priority   => 500,
   }
   -> apt::source { 'apt.postgresql.org':
-    location => $_baseurl,
-    release  => "${facts['os']['distro']['codename']}-pgdg",
-    repos    => 'main',
-    key      => {
+    location     => $_baseurl,
+    release      => "${facts['os']['distro']['codename']}-pgdg",
+    repos        => 'main',
+    architecture =>  $facts['os']['architecture'],
```

Comment:
Hum.  Real world example of a similar issue, just s/PostgreSQL/Apache/: If on an amd64 system you are using PostgreSQL with a proprietary lib which is only available for i386, you would add i386 architecture to your system (dpkg --add-architecture i386), install postgresql in its i386 flavor and be happy.

Not sure Puppet can handle installing from a specific arch that being said, but this seems perfectly valid.

Reply:
all this architecture mix is ugly :( and the upstream postgres repo doesnt even have i386 anymore.